salt/modules/glanceng.py

```diff
@@ -0,0 +1,205 @@
+# -*- coding: utf-8 -*-
+'''
+Glance module for interacting with OpenStack Glance
+
+.. versionadded:: Oxygen
+
+:depends:shade
+
+Example configuration
+
+.. code-block:: yaml
+    glance:
+      cloud: default
+
+.. code-block:: yaml
+    glance:
+      auth:
+        username: admin
+        password: password123
+        user_domain_name: mydomain
+        project_name: myproject
+        project_domain_name: myproject
+        auth_url: https://example.org:5000/v3
+      identity_api_version: 3
+'''
+
+from __future__ import absolute_import
+
+import salt.utils
+
+HAS_SHADE = False
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    pass
+
+__virtualname__ = 'glanceng'
+
+
+def __virtual__():
+    '''
+    Only load this module if shade python module is installed
+    '''
+    if HAS_SHADE:
+        return __virtualname__
+    return (False, 'The glanceng execution module failed to load: shade python module is not available')
+
+
+def compare_changes(obj, **kwargs):
+    '''
+    Compare two dicts returning only keys that exist in the first dict and are
+    different in the second one
+    '''
+    changes = {}
+    for k, v in obj.items():
+        if k in kwargs:
+            if v != kwargs[k]:
+                changes[k] = kwargs[k]
+    return changes
+
+
+def _clean_kwargs(keep_name=False, **kwargs):
+    '''
+    Sanatize the the arguments for use with shade
+    '''
+    if 'name' in kwargs and not keep_name:
+        kwargs['name_or_id'] = kwargs.pop('name')
+
+    try:
+        clean_func = __utils__['args.clean_kwargs']
+    except KeyError:
+        clean_func = salt.utils.clean_kwargs
+    return clean_func(**kwargs)
```

Comment:
Please do not use salt.utils. It will create unnecessary deprecation warnings. This entire `try` block should be rewritten as either

```python
return __utils__['args.clean_kwargs'](**kwargs)
```
or
```python
return salt.utils.args.clean_kwargs(**kwargs)
```